src/main.rs

```diff
@@ -212,31 +213,54 @@ pub async fn do_main() -> Result<()> {
 					The following is the part when the user have already subscribed
 					to an appID and now its verifying every cell that contains the data
 					*/
-					if cfg.app_id > 0 && conf >= cfg.confidence && !app_index.is_empty() {
-						for (app_id, _) in app_index.iter().filter(|app| cfg.app_id as u32 == app.0)
-						{
-							let proof =
-								rpc::get_kate_proof(&rpc_url, num, max_rows, max_cols, *app_id)
-									.await;
-							if let Ok(req_cells) = proof {
-								log::info!("\n💡Verifying all {} cells containing data of block :{} because app id {} is given ", req_cells.len(), num, app_id);
-								//hyper request for verifying the proof
-								let count = proof::verify_proof(
-									num,
-									max_rows,
-									max_cols,
-									req_cells,
-									commitment.clone(),
-								);
-								log::info!(
-									"✅ Completed {} rounds of verification for block number {} ",
-									count,
-									num
-								);
-							} else {
-								log::info!("\n ❌ getting proof cells failed, data availability cannot be ensured");
-							}
-						}
+					//@TODO : Major optimization needed here
+					if conf >= cfg.confidence && !app_index.is_empty() {
+						let query_cells: Vec<types::Cell> = match cfg.app_id as i16 {
+							n if n == -1 => {
+								let mut cells = Vec::new();
+								for i in 0..max_rows {
+									for j in 0..max_cols {
+										cells.push(types::Cell {
+											block: num,
+											row: i,
+											col: j,
+											..Default::default()
+										});
+									}
+								}
+								cells
```

Comment:
Maybe shorter version (not sure which one is better):

`(0..cols).flat_map(move |col| (0..rows).map(move |row| types::Cell { block: num, row, col, ..Default::default() });` 

Also we could move this to typed mod?

Reply:
yeh thanks! I will try it out